node/network/gossip-support/src/lib.rs

```diff
@@ -223,45 +243,54 @@ where
 					self.last_session_index = Some(session_index);
 				}
 
-				let all_authorities = determine_relevant_authorities(ctx, relay_parent).await?;
-				let our_index = ensure_i_am_an_authority(&self.keystore, &all_authorities).await?;
-				let other_authorities = {
-					let mut authorities = all_authorities.clone();
-					authorities.swap_remove(our_index);
-					authorities
-				};
+				let our_index =
+					ensure_i_am_an_authority(&self.keystore, &session_info.discovery_keys).await?;
```

Comment:
ah, good catch. I'll adjust it to make connections even if this fails.

Reply:
We probably still want to limit these to current/past/present authorities. Otherwise collators and full nodes with overseer will constantly try to connect and fail.